base58/__init__.py

```diff
@@ -74,9 +79,16 @@ def b58decode_int(
     v = v.rstrip()
     v = scrub_input(v)
 
+    map = _get_base58_decode_map(alphabet)
+
     decimal = 0
     for char in v:
-        decimal = decimal * 58 + alphabet.index(char)
+        # char_position = map[char]
```

Comment:
this commented part of code allows the code to pass external unit tests.
However it makes a code a bit more complicated and looses 5-10% of performance for short strings like BTC address.
I wonder if we need to keep original bahavior and tests or now.

Reply:
I added a test to the same effect on master, I think it's important to keep this behaviour

Reply:
I have found a very fast and simple solution to raise exception in case of wrong character.
But it may be KeyError or TypeError. I will continue searching to reproduce ValueError, but is it necessary?

Reply:
it's interesting that the map lookup would be as quick as a simple indexing into a byte array, LGTM!

ValueError is still the correct type to use here but reraising the keyerror should be good enough and the performance impact of try/except *should* be negligible in the happy case. So lets just wrap that entire loop like

```
try:
  for char in v:
    ...
except KeyError:
  raise ValueError("Invalid character") from None
```

Reply:
That is good, simple and effective solution. For some reason I was missing it :)